File: src/main/java/network/reflected/rfnetapi/purchases/PurchaseSuccessEvent.java

package network.reflected.rfnetapi.purchases;

import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

/**
 * Represents a player related inventory event
 */
public class PurchaseSuccessEvent extends Event {
    private static final HandlerList handlers = new HandlerList();
    protected Product product;
    protected Player purchaser;

    public PurchaseSuccessEvent(@NotNull Product product, @NotNull Player purchaser) {
        this.product = product;
        this.purchaser = purchaser;
    }

    public Product getPurchasedProduct() {
        return this.product;
    }

    public Player getPurchaser() {
        return this.purchaser;
    }

    @NotNull
    @Override
    public HandlerList getHandlers() {
        return handlers;
    }

    @NotNull
    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/java/network/reflected/rfnetapi/purchases/PurchaseSuccessEvent.kt
package network.reflected.rfnetapi.purchases

import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

/**
 * Represents a player related inventory event
 */
class PurchaseSuccessEvent(var purchasedProduct: Product, var purchaser: Player) : Event() {
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList = HandlerList()
    }
}